iocell/src/main/scala/chisel/IOCell.scala

```diff
@@ -0,0 +1,188 @@
+// See LICENSE for license details
+
+package barstools.iocell.chisel
+
+import chisel3._
+import chisel3.util.{Cat, HasBlackBoxResource}
+import chisel3.experimental.{Analog, DataMirror, IO}
+
+class AnalogIOCellBundle extends Bundle {
+  val pad = Analog(1.W)
+  val core = Analog(1.W)
+}
+
+class DigitalGPIOCellBundle extends Bundle {
+  val pad = Analog(1.W)
+  val i = Output(Bool())
+  val ie = Input(Bool())
+  val o = Input(Bool())
+  val oe = Input(Bool())
+}
+
+class DigitalOutIOCellBundle extends Bundle {
+  val pad = Output(Bool())
+  val o = Input(Bool())
+  val oe = Input(Bool())
+}
+
+class DigitalInIOCellBundle extends Bundle {
+  val pad = Input(Bool())
+  val i = Output(Bool())
+  val ie = Input(Bool())
+}
+
+abstract class IOCell extends BlackBox with HasBlackBoxResource
+
+abstract class AnalogIOCell extends IOCell {
+  val io: AnalogIOCellBundle
+}
+
+abstract class DigitalGPIOCell extends IOCell {
+  val io: DigitalGPIOCellBundle
+}
+
+abstract class DigitalInIOCell extends IOCell {
+  val io: DigitalInIOCellBundle
+}
+
+abstract class DigitalOutIOCell extends IOCell {
+  val io: DigitalOutIOCellBundle
+}
+
+class ExampleAnalogIOCell extends AnalogIOCell {
+  val io = IO(new AnalogIOCellBundle)
+  addResource("/barstools/iocell/vsrc/IOCell.v")
```

Comment:
Should we just move this into the abstract class?

Reply:
No, because I want users to be able to extend the abstract class with their own implementations, which would have a different resource. I thought about making an `ExampleIOCell` trait (can name it something else) that does this but it's about the same amount of characters, so it doesn't buy much.

Reply:
Hmm...ok. I think we should discuss extension use cases more concretely (and write documentation about how they would work).

Reply:
I added some explanation inline... if it's insufficient let's chat more.